FirebaseMLModelDownloader/Sources/ModelInfoRetriever.swift

```diff
@@ -34,38 +34,43 @@ extension ModelInfoResponse {
 
 /// Model info retriever for a model from local user defaults or server.
 class ModelInfoRetriever: NSObject {
-  /// Current Firebase app options
+  /// Current Firebase app options.
   private var options: FirebaseOptions
-  /// Model info associated with model.
-  var modelInfo: ModelInfo?
   /// Model name.
-  var modelName: String
+  private var modelName: String
   /// Firebase installations.
-  var installations: Installations
+  private var installations: Installations
+  /// Current Firebase app name.
+  private let appName: String
+  /// Local model info to validate model freshness.
+  private var localModelInfo: LocalModelInfo?
 
   /// Associate model info retriever with current Firebase app, and model name.
-  init(modelName: String, options: FirebaseOptions, installations: Installations) {
+  init(modelName: String, options: FirebaseOptions, installations: Installations, appName: String,
+       localModelInfo: LocalModelInfo? = nil) {
     self.modelName = modelName
     self.options = options
     self.installations = installations
+    self.appName = appName
+    self.localModelInfo = localModelInfo
   }
 }
 
 /// Extension to handle fetching model info from server.
 extension ModelInfoRetriever {
   /// HTTP request headers.
-  static let fisTokenHTTPHeader = "x-goog-firebase-installations-auth"
-  static let hashMatchHTTPHeader = "if-none-match"
-  static let bundleIDHTTPHeader = "x-ios-bundle-identifier"
+  private static let fisTokenHTTPHeader = "x-goog-firebase-installations-auth"
```

Comment:
I'm curious if we mark the entire extension as `private` like `extension ModelInfoRetriever` can we avoid marking each single member?

Reply:
I did think about doing that! But I also wanted to test downloadModelInfo and I couldn't figure out how to do that without keeping it internal. Since I have a sample app now (future PR), do you think it's ok to remove the integration test?

Reply:
I did think about doing that! But I also wanted to test downloadModelInfo and I couldn't figure out how to do that without keeping it internal. Since I have a sample app now (future PR), do you think it's ok to remove the integration test?

Reply:
I would prefer to keep the tests even if it means a bit more code. I'll also take a look at the tests more carefully, maybe I can suggest anything nicer.